pkg/freeze/containerd/pause_test.go

```diff
@@ -0,0 +1,124 @@
+package containerd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"google.golang.org/grpc"
+	cri "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"knative.dev/container-freezer/pkg/daemon"
+)
+
+type FakeContainerdCRI struct {
+	paused     []string
+	resumed    []string
+	containers []*cri.Container
+	results    []string
+	method     string
+}
+
+// Container creates a CRI container with the given container ID and name
+func Container(id, name string) *cri.Container {
+	return &cri.Container{
+		Id: id,
+		Metadata: &cri.ContainerMetadata{
+			Name: name,
+		},
+	}
+}
+
+func (c *FakeContainerdCRI) List(ctx context.Context, conn *grpc.ClientConn, podUID string) (*cri.ListContainersResponse, error) {
+	return &cri.ListContainersResponse{Containers: c.containers}, nil
+}
+
+func (c *FakeContainerdCRI) Pause(ctx context.Context, ctrd *containerd.Client, container string) error {
+	c.paused = append(c.paused, container)
+	if c.method != "freeze" {
+		return fmt.Errorf("wrong method, expected: %s, got: %s", "freeze", c.method)
+	}
+	return nil
+}
+
+func (c *FakeContainerdCRI) Resume(ctx context.Context, ctrd *containerd.Client, container string) error {
+	c.resumed = append(c.resumed, container)
+	if c.method != "thaw" {
```

Comment:
Instead of asserting here let's move the assertion in to the test (and just store what we're called with in c.method). That should make it a bit clearer in the test what's going on without having to read the fake to understand things